darwin/DarwinProcess.c

```diff
@@ -9,13 +9,63 @@ in the source distribution for its full text.
 
 #include "darwin/DarwinProcess.h"
 
+#if !TARGET_OS_IPHONE
 #include <libproc.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <mach/mach.h>
 #include <sys/dirent.h>
 
+/* iOS 6.1 SDK doesn't have libproc.h, define stubs */
+#if TARGET_OS_IPHONE
+#define PROC_PIDVNODEPATHINFO 9
+#define PROC_PIDTASKINFO 4
+#define PROC_PIDTASKINFO_SIZE sizeof(struct proc_taskinfo)
+#define PROC_PIDPATHINFO_MAXSIZE 4096
+
+struct proc_taskinfo {
+   uint64_t  pti_virtual_size;
+   uint64_t  pti_resident_size;
+   uint64_t  pti_total_user;
+   uint64_t  pti_total_system;
+   uint64_t  pti_threads_user;
+   uint64_t  pti_threads_system;
+   int32_t   pti_policy;
+   int32_t   pti_faults;
+   int32_t   pti_pageins;
+   int32_t   pti_cow_faults;
+   int32_t   pti_messages_sent;
+   int32_t   pti_messages_received;
+   int32_t   pti_syscalls_mach;
+   int32_t   pti_syscalls_unix;
+   int32_t   pti_csw;
+   int32_t   pti_threadnum;
+   int32_t   pti_numrunning;
+   int32_t   pti_priority;
+};
+
+struct vnode_info_path {
+   char vip_path[1024];
+};
+
+struct proc_vnodepathinfo {
+   struct vnode_info_path pvi_cdir;
+   struct vnode_info_path pvi_rdir;
+};
+
+static inline int proc_pidpath(int pid, void *buffer, uint32_t buffersize) {
+   (void)pid; (void)buffer; (void)buffersize;
+   return -1;
+}
+
+static inline int proc_pidinfo(int pid, int flavor, uint64_t arg, void *buffer, int buffersize) {
+   (void)pid; (void)flavor; (void)arg; (void)buffer; (void)buffersize;
+   return -1;
+}
```

Comment:
If these are implemented as stubs, you shouldn't be defining the PROC_EXE and CWD columns for the platform either. And as those two columns are kinda essential these information are kinda essential for an implementation to provide.

Reply:
No longer applicable. Fixed.